Fixes unexpected NSI matches

mudit06mah opened the pull request:
### Issue Fixed:
#11215 

### Issue:
**Before:**
NSI was suggesting tag upgradation for "OK Plus" with "Circle K"
<img width="412" height="508" alt="image" src="https://github.com/user-attachments/assets/9814591a-126c-4311-81a8-9632d2964bb0" />

**After:**
No such error is suggested.

### PR Explanation:
When gathering all the names, for example in "OK Plus"
A tuple was generated for "OK Plus" as well as "OK".
When comparing "ok" with "circlek", we obtain a match (I believe it happens because "ok" is a very short string)
I fixed it by directly comparing the complete search string so that it doesn't accidentally match with strings it isn't completely similar to.

### Feedback:
I could not understand why we're using `gatherNames()` and `gatherTuples()`.

mudit06mah committed:
nsi matcher fix

mudit06mah committed:
Fixed linting error

bhousel commented:
> Still here's a fix for this Issue.

How is this PR supposed to fix the issue?

mudit06mah commented:
> > Still here's a fix for this Issue.
> 
> How is this PR supposed to fix the issue?

The tag upgradation was being suggested because "OK" was a really small string and was somehow matching with "circlek", so I skipped all strings with size less than 3 
, thus removing unwanted suggestions.

If you think a better method is to match the string before skipping it, then I will add that

mudit06mah commented:
I just realised why this is not a good fix.
I apologise for the inconvenience and I will fix it first thing tomorrow

mudit06mah committed:
Fixed nsi issue by directly comparing short strings

mudit06mah commented:
I have fixed my PR by comparing small strings instead of checking, (so for example, name like "kfc" will not be skipped).
Hence this issue has been fixed without causing any more problems.

bhousel commented:
Why not just add the "OK Plus" brand to NSI?

These are the "OK" branded convenience stores we are tracking already:
https://nsi.guide/index.html?t=brands&k=shop&v=convenience&tt=ok
That page lists the different name tags that will match, and which locations around the world that it can be expected to be found.

mudit06mah commented:
> Why not just add the "OK Plus" brand to NSI?
> 
> These are the "OK" branded convenience stores we are tracking already: https://nsi.guide/index.html?t=brands&k=shop&v=convenience&tt=ok That page lists the different name tags that will match, and which locations around the world that it can be expected to be found.

I think the issue here is that the current algorithim is forcefully matching brands like "OK Plus" and "Circle K" which are completely different names, and hence I fixed that.
I didn't know that the brand "OK" is already being tracked. But even if it is, I fixed the code to not match different names like "OK Plus" and "Circle K".

I will add "OK Plus" to the NSI list if it is still needed.

UKChris-osm commented:
It seems [Circle K](https://nsi.guide/index.html?t=brands&k=shop&v=convenience&tt=ok#circlek-04f01c) includes a `matchName` for **ok**, but I'm not sure why - perhaps we could also remove this `matchName` to stop the suggestion?

bhousel commented:
> It seems [Circle K](https://nsi.guide/index.html?t=brands&k=shop&v=convenience&tt=ok#circlek-04f01c) includes a `matchName` for **ok**, but I'm not sure why - perhaps we could also remove this `matchName` to stop the suggestion?

Yeah I agree, it seems like we should 
1. Remove the "ok" and "ok-mart" matchNames for the worldwide Circle K.  
2. Review existing items for "OK" branded stores in the parts of the world where they have it (China, Hong Kong, Macau, also  a Chilean and a South African brands that seems unrelated).
3. Add a new item for "OK Plus" scoped to wherever that is used.

I opened https://github.com/osmlab/name-suggestion-index/issues/11345
